examples/base/package.json

```diff
@@ -0,0 +1,37 @@
+{
+  "name": "@raincatcher/base",
+  "version": "1.0.0",
+  "description": "NPM package template for the RainCatcher project",
+  "main": "index.js",
+  "author": "feedhenry-raincatcher@redhat.com",
+  "license": "Apache-2.0",
+  "scripts": {
+    "build": "del lib && tsc",
+    "test": "nyc mocha",
+    "prepublish": "npm run build"
+  },
+  "nyc": {
+    "include": [
+      "src/**/*.ts"
+    ],
+    "extension": [
+      ".ts"
+    ],
+    "require": [
+      "ts-node/register"
+    ],
+    "check-coverage": true,
+    "lines": 75,
+    "functions": 100,
+    "branches": 80
+  },
+  "devDependencies": {
+    "@types/mocha": "^2.2.41",
+    "del-cli": "^1.0.0",
+    "mocha": "^3.4.2",
+    "nyc": "^11.0.1",
+    "source-map-support": "^0.4.15",
```

Comment:
source-map-support is recommended in the [istanbul instructions for typescript](https://istanbul.js.org/docs/tutorials/typescript/), but I think it just affects compiled TS dependencies so we can have a good stack trace on them. Since this setup is running the unit tests with `ts-node` it automatically points to the right TS source line for exceptions thrown from the module itself.

TS already outputs source maps by enabling an extra flag that I've already did on the tsconfig

We have three alternatives for the unit test setup:

- Run own module with [`ts-node`](https://github.com/TypeStrong/ts-node). But since we have to point `package.json#main` to `lib/index.js` it'll still run internal dependencies as compiled, so we might need to recompile them before running (trivial with lerna, but not so fast)
- Run unit tests on own module on top of compiled js and rely on source map support. That way there's no difference between the module being tested and its dependencies.
- Figure out a way to require all ts sources for unit testing (maybe by modifiying bootstrapping when NODE_ENV=test). That way `ts-node` runs a full TS stack and it's much faster to iterate on changes to dependency modules too.

What's your recommendation? also ping @wtrocki